sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/RxCollectionCache.java

```diff
@@ -29,9 +31,18 @@
     private final AsyncCache<String, DocumentCollection> collectionInfoByNameCache;
     private final AsyncCache<String, DocumentCollection> collectionInfoByIdCache;
 
+    public static void serialize(CosmosClientState state, RxCollectionCache cache) throws IOException {
+        state.serializeCollectionInfoByIdCache(cache.collectionInfoByIdCache);
+        state.serializeCollectionInfoByNameCache(cache.collectionInfoByNameCache);
```

Comment:
By name is a reference or derived one. 

How about limit to primitive data only?

Reply:
this is by a reference. serializing `cache.collectionInfoByIdCache` and `cache.collectionInfoByIdCache` are necessary for taking a snapshot of the CollectionCache.